File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    def dfs(index, current_sum):
        # 모든 숫자를 다 처리한 경우
        if index == len(numbers):
            return 1 if current_sum == target else 0
        
        # 현재 숫자에 대해 +와 -를 각각 붙여서 재귀 호출
        return dfs(index + 1, current_sum + numbers[index]) + dfs(index + 1, current_sum - numbers[index])

    return dfs(0, 0)

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    const length = numbers.length;
    let answer = 0;
    
    function dfs(index, currentSum) {
        if(index === length) {
            if(currentSum === target) {
                answer++;
            }
            return;
        }
        
        const nextIndex = index + 1;
        const number = numbers[index];
 
        dfs(nextIndex, currentSum + number);
        dfs(nextIndex, currentSum - number);
    }
    
    dfs(0,0);
    
    return answer;
}